pkgs/by-name/mc/mcpelauncher-client/package.nix

```
@@ -0,0 +1,117 @@
+{
+  lib,
+  callPackage,
+  clangStdenv,
+  fetchFromGitHub,
+  fetchpatch,
+  cmake,
+  pkg-config,
+  openssl,
+  zlib,
+  libpng,
+  libglvnd,
+  xorg,
+  libevdev,
+  curl,
+  pulseaudio,
+  qt6,
+  glfw,
+  withQtWebview ? true,
+  withQtErrorWindow ? true,
+}:
+
+# gcc doesn't support __has_feature
+clangStdenv.mkDerivation (finalAttrs: {
+  pname = "mcpelauncher-client";
+  version = "1.1.2-qt6";
+
+  src = fetchFromGitHub {
+    owner = "minecraft-linux";
+    repo = "mcpelauncher-manifest";
+    tag = "v${finalAttrs.version}";
+    fetchSubmodules = true;
+    hash = "sha256-PmCq6Zgtp17UV0kIbNouFwj/DMiTqwE31+tTb2LUp5o=";
+  };
+
+  patches = [
+    ./dont_download_nlohmann_json.patch
+    ./dont_download_glfw_client.patch
+    # These are upcoming changes that have been merged upstream. Once these get in a release, remove these patches.
+    (fetchpatch {
+      url = "https://github.com/minecraft-linux/game-window/commit/feea8c0e0720eea7093ed95745c17f36d6c40671.diff";
+      hash = "sha256-u4uveoKwwklEooT+i+M9kZ0PshjL1IfWhlltmulsQJo=";
+      stripLen = 1;
+      extraPrefix = "game-window/";
+    })
+    (fetchpatch {
+      url = "https://github.com/minecraft-linux/mcpelauncher-client/commit/db9c31e46d7367867c85a0d0aba42c8144cdf795.diff";
+      hash = "sha256-za/9oZYwKCYyZ1BXQ/zeEjRy81B1NpTlPHEfWAOtzHk=";
+      stripLen = 1;
+      extraPrefix = "mcpelauncher-client/";
+    })
+  ];
+
+  # FORTIFY_SOURCE breaks libc_shim and the project will fail to compile
+  hardeningDisable = [ "fortify" ];
+
+  nativeBuildInputs =
+    [
+      cmake
+      pkg-config
+    ]
+    ++ lib.optionals (withQtWebview || withQtErrorWindow) [
+      qt6.wrapQtAppsHook
+    ];
+
+  buildInputs =
+    [
+      openssl
+      finalAttrs.nlohmann_json_373
+      zlib
+      libpng
+      libglvnd
+      xorg.libX11
+      xorg.libXi
+      xorg.libXtst
+      libevdev
+      curl
+      pulseaudio
+      qt6.qttools
+      glfw
+    ]
+    ++ lib.optionals withQtWebview [
+      qt6.qtwebengine
+    ];
+
+  nlohmann_json_373 = callPackage ./nlohmann_json_373_dep.nix { };
+
+  cmakeFlags =
+    [
+      "-DUSE_OWN_CURL=OFF"
+      "-DENABLE_DEV_PATHS=OFF"
+      "-Wno-dev"
+      "-DCMAKE_BUILD_TYPE=Release"
+      "-DGAMEWINDOW_SYSTEM=GLFW"
+      "-DUSE_SDL3_AUDIO=OFF"
+    ]
+    ++ lib.optionals (!withQtWebview) [
+      "-DBUILD_WEBVIEW=OFF"
+      "-DXAL_WEBVIEW_USE_CLI=ON"
+      "-DXAL_WEBVIEW_USE_QT=OFF"
+    ]
+    ++ lib.optionals (!withQtErrorWindow) [
+      "-DENABLE_QT_ERROR_UI=OFF"
+    ];
+
+  meta = {
+    description = "Unofficial *NIX launcher of Minecraft: Bedrock Edition using Android - CLI launcher";
```

Feels a bit wordy IMO

```suggestion
    description = "Unofficial *NIX launcher of Minecraft: Bedrock Edition";
```